bittensor/core/async_subtensor.py

```diff
@@ -369,6 +446,56 @@ async def get_balance(
             results.update({item[0].params[0]: Balance(value["data"]["free"])})
         return results
 
+    async def get_transfer_fee(
+        self, wallet: "Wallet", dest: str, value: Union["Balance", float, int]
+    ) -> "Balance":
+        """
+        Calculates the transaction fee for transferring tokens from a wallet to a specified destination address. This function simulates the transfer to estimate the associated cost, taking into account the current network conditions and transaction complexity.
+
+        Args:
+            wallet (bittensor_wallet.Wallet): The wallet from which the transfer is initiated.
+            dest (str): The ``SS58`` address of the destination account.
+            value (Union[bittensor.utils.balance.Balance, float, int]): The amount of tokens to be transferred, specified as a Balance object, or in Tao (float) or Rao (int) units.
+
+        Returns:
+            bittensor.utils.balance.Balance: The estimated transaction fee for the transfer, represented as a Balance object.
+
+        Estimating the transfer fee is essential for planning and executing token transactions, ensuring that the wallet has sufficient funds to cover both the transfer amount and the associated costs. This function provides a crucial tool for managing financial operations within the Bittensor network.
+        """
+        if isinstance(value, float):
+            value = Balance.from_tao(value)
+        elif isinstance(value, int):
+            value = Balance.from_rao(value)
+
+        if isinstance(value, Balance):
+            call = await self.substrate.compose_call(
+                call_module="Balances",
+                call_function="transfer_allow_death",
+                call_params={"dest": dest, "value": value.rao},
+            )
+
+            try:
+                payment_info = await self.substrate.get_payment_info(
+                    call=call, keypair=wallet.coldkeypub
+                )
+            except Exception as e:
+                logging.error(
+                    f":cross_mark: <red>Failed to get payment info: </red>{e}"
+                )
+                payment_info = {"partialFee": int(2e7)}  # assume  0.02 Tao
+
+            fee = Balance.from_rao(payment_info["partialFee"])
+            return fee
+        else:
+            fee = Balance.from_rao(int(2e7))
+            logging.error(
+                "To calculate the transaction fee, the value must be Balance, float, or int. Received type: %s. Fee "
+                "is %s",
+                type(value),
+                2e7,
+            )
+            return fee
+
```

Comment:
This function will become so much cleaner if you just specify that the value is of `Balance` type and then you don't check if it actually is. Maybe they subclassed it, maybe they implemented the same interface in some other way, or maybe they passed in something different and they'll get an exception - anyway it's their fault as they didn't pass the proper arguments to a function call.

Reply:
I will look deeper into what this might affect and apply the changes. 
Thanks for this call out!